x/ibc/light-clients/07-tendermint/types/client_state.go

```diff
@@ -429,6 +469,20 @@ func (cs ClientState) VerifyNextSequenceRecv(
 	return nil
 }
 
+func verifyDelayPeriodPassed(store sdk.KVStore, proofHeight exported.Height, currentTimestamp, delayPeriod uint64) error {
+	// check that executing chain's timestamp has passed consensusState's processed time + delay period
+	processedTime, ok := GetProcessedTime(store, proofHeight)
+	if !ok {
+		return sdkerrors.Wrapf(ErrProcessedTimeNotFound, "processed time not found for height: %s", proofHeight)
+	}
+	validTime := processedTime + delayPeriod
+	if validTime > currentTimestamp {
```

Comment:
we should document that this is inclusive ie a valid time is >= current timestamp